src/server/license/server/api_server.go

```diff
@@ -390,3 +392,140 @@ func (a *apiServer) ListUserClusters(ctx context.Context, req *lc.ListUserCluste
 		Clusters: clusters,
 	}, nil
 }
+
+func (a *apiServer) RevokeTokensForUserAcrossClusters(ctx context.Context, req *lc.RevokeTokensForUserAcrossClustersRequest) (resp *lc.RevokeTokensForUserAcrossClustersResponse, retErr error) {
+	a.LogReq(req)
+	defer func(start time.Time) { a.pachLogger.Log(req, resp, retErr, time.Since(start)) }(time.Now())
+
+	result, err := a.clusterFanout(ctx, req.IdToken, func(pachClient *client.APIClient) error {
+		_, err := pachClient.AuthAPIClient.RevokeAuthTokensForUser(pachClient.Ctx(), &auth.RevokeAuthTokensForUserRequest{Username: req.Username})
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	// transform FanoutResult into a Response pb
+	failures := make([]*lc.DelegatedCallFailure, len(result.failures))
+	for i, r := range result.failures {
+		failures[i] = &lc.DelegatedCallFailure{ReceiverId: r.id, Err: r.err.Error()}
+	}
+	return &lc.RevokeTokensForUserAcrossClustersResponse{
+		SuccessClusters:  result.successes,
+		FailureClusters:  failures,
+		TimedoutClusters: result.timeouts,
+	}, nil
+}
+
+type fanoutError struct {
+	id  string
+	err error
+}
+
+type fanoutReport struct {
+	successes []string
+	failures  []*fanoutError
+	timeouts  []string
+}
+
+// cluster fanout uses an ID Token permitted to make requests on behalf of the User to Authenticate to each cluster
+func (a *apiServer) clusterFanout(ctx context.Context, idToken string, cb func(*client.APIClient) error) (*fanoutReport, error) {
+	res, err := a.ListClusters(ctx, &lc.ListClustersRequest{})
+	if err != nil {
+		return nil, err
+	}
+	return ClusterFanout(ctx, res.Clusters, idToken, cb)
+}
+
+// breaking out this public for now in an effort to test it
+func ClusterFanout(ctx context.Context, clusters []*lc.ClusterStatus, idToken string, cb func(*client.APIClient) error) (*fanoutReport, error) {
+	waitingOnClusters := make(map[string]bool) // a set of IDs for clusters yet to respond
+	successChan := make(chan string)
+	failureChan := make(chan *fanoutError)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	for _, cluster := range clusters {
+		waitingOnClusters[cluster.Id] = true
+
+		// kickoff goroutine to make request to a pachd cluster
+		go func(ctx context.Context, address string, id string) {
```

Comment:
I think this could be simplified with an error group (https://pkg.go.dev/golang.org/x/sync/errgroup). Since we propagate the context thorugh all the RPCs we could set a timeout on the context and then we don't have to worry about enforcing those timeouts ourselves either.